torch_geometric/distributed/dist_neighbor_sampler.py

```diff
@@ -620,12 +619,12 @@ def _sample_one_hop(
             True,  # csc
             self.replace,
             self.subgraph_type != SubgraphType.induced,
-            self.disjoint and self.time_attr is not None,
+            self.disjoint or self.time_attr is not None,
```

Comment:
Hi @rusty1s, thank you for the review. One comment from me about this change: during distributed training, layer-by-layer sampling loop is located in Python and not in the sample() function from pyg-lib, therefore, in the next layers, information about nodes belonging to the subgraph is lost in the c++ code and the batch calculated in the C++ is not important. Because of that the whole pipeline was created such that the sample() function must work in disjoint mode only during temporal sampling. If there is only a disjoint, the sample() function does not have to use this option because the batch is counted later anyway based on the sampled nodes with duplicates info and the one from C++ is not used anyway. So this condition should stay as it was.

Reply:
Thanks for the clarification. I can change that back. Why do we then expose `disjoint` as an argument to `DistNeighborSampler`?

Reply:
https://github.com/pyg-team/pytorch_geometric/pull/8482

Reply:
This is what the algorithm looks like:

1. Subgraph ids are assigned to the seed nodes at the beginning before layer-by-layer sampling starts in the node_sample() function in the DistNeighborSampler:
`src_batch = torch.arange(batch_size) if self.disjoint else None`

2. src_batch is being passed to the sample_one_hop() function where all machines starts sampling.
3. When all machines finished sampling, their results must be merged - it is being done in the merge_sampler_outputs().
4. In the merge_sampler_outputs() function, based on the sampled nodes with duplicates and information about how many neighbors each node has sampled, subgraph ids are assigned to the newly sampled nodes and out_batch is being returned.
5. Then, in the next layer, we assign the newly created batch (of the newly sampled nodes) to the src_batch variable and go back to the bullet 2.

Reply:
Got it, thanks for clarifying :)